connector_extension/components/adapter.py

```diff
@@ -16,6 +16,13 @@ class ConnectorExtensionAdapterCRUD(AbstractComponent):
 
     _date_format = "%Y-%m-%d"
     _datetime_format = "%Y-%m-%dT%H:%M:%SZ"
+    _datetimestamp_format = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+    def _str_to_datetime(self, sdt):
+        try:
+            return datetime.datetime.strptime(sdt, self._datetimestamp_format)
+        except ValueError:
+            return datetime.datetime.strptime(sdt, self._datetime_format)
```

Comment:
This is not generic enough to be at connector_extension, please put it back to the specific connector